[react-native] Add a basic sample for AppConfiguration (#21907)
Azure/azure-sdk-for-js: samples/frameworks/react-native-expo/ts/appconfigBasic/README.md

# Azure SDK samples for React Native using Expo (TypeScript)

This sample application shows how to use the TypeScript client libraries for Azure in some common scenarios.

> In this sample, we build a simple application in React Native using Expo and integrating with Azure Application Configuration service.

Reviewer: ```suggestion
In this sample, we build a simple application in React Native using Expo and integrating with Azure App Configuration service.
```

## Prerequisites

The samples are compatible with [LTS versions of Node.js](https://nodejs.org/about/releases/).

Before running the samples in Node, they must be compiled to JavaScript using the TypeScript compiler. For more information on TypeScript, see the [TypeScript documentation][typescript].

The sample is using [Expo](https://expo.dev/), a framework and platform for universal React applications.

You need [an Azure subscription][freesub] and the following resources created to run this sample:

## Step by step

### Create the project

Run expo CLI to create a blank project with typescript support, give it a name of `appconf-rn-expo`.

```shell
expo init -t expo-template-blank-typescript
```

> At this point we should be ale to see the app running if the environment is set up correctly

Reviewer: ```suggestion
At this point, we should be able to see the app running if the environment is set up correctly
```

...

      <Text>Open up App.tsx to start working on your app!</Text>
+     <Button title="Click me!" onPress={() => runAppConfigSample() } />
      <StatusBar style="auto" />
```

### Add the App Configuration sample file

create a new file `appConfigSample.ts` in the sample directory as `App.tsx` with following content from [appConfigSample.ts](./appConfigSample.ts) file

### Add dependencies

We need to add a dependency to the JavaScript Client SDK library for Azure App Configuration (version 1.4.0-beta.1 or later). We use `@babel/plugin-proposal-async-generator-functions` to help transform async iterator usage in our sample. We also use `babel-plugin-inline-dotenv` to help load secrets from our `.env` file while developing.

```shell
yarn add @azure/app-configuration@1.4.0-beta.1
yarn add --dev babel-plugin-inline-dotenv @babel/plugin-proposal-async-generator-functions
```

Then add the following into `babel.config.js` to enable the plugin

```diff
    presets: ["babel-preset-expo"],
+    plugins: [
+      "@babel/plugin-proposal-async-generator-functions",
+      ["inline-dotenv", { unsafe: true }],
+    ],
```

### Add connection string to .env file

> Create a `.env` file in the project directory. Retrieve your connection string from Azure portal and add them into the `.env` file. Since this file contains secrets you need to add it to the ignore list if your code is commit into a repository.

Reviewer: ```suggestion
Create a `.env` file in the project directory. Retrieve your connection string from Azure portal and add them to the `.env` file. Since this file contains secrets you need to add it to the ignore list if your code is committed to a repository.
```
